proof/refine/ARM/VSpace_R.thy

```diff
@@ -883,7 +883,13 @@ lemma deleteASID_corres:
        apply (simp add: mask_asid_low_bits_ucast_ucast)
       apply wp
       apply (simp add: o_def)
-      apply (wp getASID_wp)
+      \<comment> \<open>FIXME: This proof weakens the schematic precondition so that an extra variable can be used,
+                and then uses `(clarsimp, assumption)` to resolve the schematic.
+                An alternative would be to use @{thm Some_to_the} to rewrite the conclusion, so that
+                the precondition does not need to depend on that variable.
+                Both approaches feel fragile.\<close>
+      apply (rule hoare_pre)
+       apply (wp getASID_wp)
```

Comment:
At least it does this explicitly now. I'd slightly favour the `Some_to_the` approach to be honest. At least that would likely fail fast when something changes, whereas an explicit schematic and then `assumption` might do weird things.